sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/UriDescribedType.java

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus.implementation;
+
+import java.net.URI;
+
+import static com.azure.messaging.servicebus.implementation.ServiceBusConstants.URI_SYMBOL;
+
+/**
+ * URI described type.
+ */
+public class UriDescribedType extends ServiceBusDescribedType {
+
+    /**
+     * Set described to describe data in described type.
+     *
+     * @param described  real value in the described type.
+     */
+    public UriDescribedType(Object described) {
+        super(URI_SYMBOL, ((URI) described).toString());
+    }
+
+    @Override
+    public int size() {
+        return URI_SYMBOL.length() + ((String) this.getDescribed()).length();
```

Comment:
This part seems not correct `((String) this.getDescribed()).length()`? It is char length, not encoded length.

Reply:
Yeah, sorry, I think URI only contains ASCII at that time, but it may contain other String. It should use `this.getDescribed().getBytes(StandardCharsets.UTF_8).length`